homeassistant/components/marstek/__init__.py

```diff
@@ -0,0 +1,294 @@
+"""The Marstek integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from pymarstek import MarstekUDPClient, get_es_mode
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.device_registry import format_mac
+from homeassistant.helpers.typing import ConfigType
+
+from .const import DEFAULT_UDP_PORT, DOMAIN, PLATFORMS
+from .coordinator import MarstekDataUpdateCoordinator
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+type MarstekConfigEntry = ConfigEntry[
+    tuple[MarstekUDPClient, MarstekDataUpdateCoordinator, dict[str, Any]]
```

Comment:
Can you create a dataclass for these values? This way we will know what these fields mean.
https://github.com/home-assistant/core/blob/c9351a022ea28546afface55f2bc7d9d0d8de7d7/homeassistant/components/accuweather/coordinator.py#L39-L45